conda/gateways/disk/update.py

```diff
@@ -62,6 +62,11 @@ def rename(source_path, destination_path, force=False):
             os_rename(source_path, destination_path)
         except EnvironmentError as e:
             if e.errno in (EINVAL, EXDEV):
+                # https://github.com/conda/conda/issues/6811
+                # Don't attempt fallback for directories, let the caller catch EXDEV, see:
+                # https://github.com/conda/conda/blob/4.4.9/conda/core/path_actions.py#L1243-L1254
+                if e.errno == EXDEV and isdir(source_path):
+                    raise
```

Comment:
To be honest, I think we should just get rid of this fallback and add EINVAL to https://github.com/conda/conda/blob/4.4.9/conda/core/path_actions.py#L1246

But I am not sure of rename is being called only from that place or not.

Reply:
cc @quasiben

Reply:
`rename()` is called 7 times through `backoff_rename()` in `conda/core/path_actions.py`. It looks like that's currently the full use of it.

Reply:
When using `backoff_rename()` in a `reverse()` method, I don't think we have to worry about any special handling for docker filesystem issues.

Use of `backoff_rename()` in `execute()` methods at:

* https://github.com/conda/conda/blob/4.4.9/conda/core/path_actions.py#L974
* https://github.com/conda/conda/blob/4.4.9/conda/core/path_actions.py#L1151
* https://github.com/conda/conda/blob/4.4.9/conda/core/path_actions.py#L1244

Reply:
> I think we should just get rid of this fallback completely and add EINVAL to 

I'm ok with raising the error handling up to the caller of `rename()`/`backoff_rename()` for now.  @nehaljwani Why don't you go ahead and see what that looks like.  Just make sure to add notes in top-line comments for `rename()` and `backoff_rename()` about the errors future users of those functions might want to think about handling.